### PR TITLE
Migrate to sbt-typelevel

### PR DESCRIPTION
**Pull request by armanbilge**
Yeah yeah I know, this needs a backport to the other branch as well.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -88,65 +32,43 @@ val logbackClassicV = "1.2.10"
...
     name := "log4cats-noop"
   )
-lazy val noopJVM = noop.jvm
-lazy val noopJS = noop.js
 
 lazy val slf4j = project
-  .settings(commonSettings, releaseSettings)
-  .dependsOn(coreJVM)
+  .settings(commonSettings)
+  .dependsOn(core.js)
```
Comment (danicheg) [inline build.sbt:71]: Not a blocker at all, just curious, why did you change the `coreJVM` to `core.js` here?

**Review by danicheg — APPROVED** [review]

File: build.sbt [hunk 2]
```diff
@@ -88,65 +32,43 @@ val logbackClassicV = "1.2.10"
...
-  .settings(commonSettings, releaseSettings)
+  .settings(commonSettings)
   .settings(
     name := "log4cats-core",
     libraryDependencies ++= Seq(
       "org.typelevel" %%% "cats-core" % catsV
     )
   )
-lazy val coreJVM = core.jvm
-lazy val coreJS = core.js
```
Comment (armanbilge) [inline build.sbt:123]: I deleted `coreJVM` and friends here. They are effectively redundant.

File: build.sbt (hunk 1 above)
Comment (armanbilge) [reply build.sbt:71]: See my comment https://github.com/typelevel/log4cats/pull/592/files#r794407004. It's a style of code I usually see in legacy builds.

**Comment by danicheg** [comment]
I think this is pretty good to go onward. We should also backport this for the branch with CE 2. I could tackle on, but it may take some time.
